qiskit_aqua/utils/run_circuits.py

```diff
@@ -271,7 +298,10 @@ def compile_and_run_circuits(circuits, backend, backend_config, compile_config,
                     qobj = qobjs[idx]
                     #  assure job get its id
                     while True:
-                        job = backend.run(qobj, **backend_options, **noise_config)
+                        job = run_on_backend(backend, qobj,
```

Comment:
may I know at what situation the `job._future` is not `None` in `run_on_backend`?
if `run_on_backend` could possibly raise error, we might need try and except block around this call (and the above one).

Reply:
Good catch, there is none. I was going off of the job submission code, which is different. Removed the condition.